File: app/src/main/java/com/lcl/lclmeasurementtool/Database/Entity/ConnectivityViewModel.java
package com.lcl.lclmeasurementtool.Database.Entity;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.lcl.lclmeasurementtool.Database.DB.MeasurementResultDatabase;

import java.util.List;

/**
 * The connectivity measurement view model interfacing between database and the view
 */
public class ConnectivityViewModel extends AbstractViewModel<Connectivity> {

    private final MeasurementResultDatabase db;
    public ConnectivityViewModel(@NonNull Application application) {
        super(application);
        db = MeasurementResultDatabase.getInstance(application);
    }

    @Override
    public LiveData<List<Connectivity>> getAll() {
        return db.connectivityDAO().retrieveAllConnectivities();
    }

    @Override
    public void insert(Connectivity data) {
        MeasurementResultDatabase.databaseWriteExecutor.execute(() -> {
            Log.d("Connectivity_ViewModel", "[insert called]");
            db.connectivityDAO().insert(data);
        });
    }
}

File: app/src/main/java/com/lcl/lclmeasurementtool/model/viewmodel/ConnectivityViewModel.kt
package com.lcl.lclmeasurementtool.model.viewmodel

import androidx.lifecycle.ViewModel
import com.lcl.lclmeasurementtool.model.dao.ConnectivityDao
import com.lcl.lclmeasurementtool.model.datamodel.ConnectivityReportModel

class ConnectivityViewModel(private val connectivityDao: ConnectivityDao): ViewModel() {
    fun fullConnectivityList(): List<ConnectivityReportModel> = connectivityDao.getAll()
}